### PR TITLE
Update consensus mechanisms docs

### PR DESCRIPTION
**Pull request by b17z**
## Description

Additions to the consensus mechanism docs

## Changes in Detail

- Adds notes around Byzantine fault and Byzantine Generals Problem as it provides more context on consensus and consensus mechanisms
- Adds another an analogy/story to support the Byzantine Generals Problem explanation
- Adds a note on the difference between protocols and algorithms

## Related Issue

  #5711

### CODE REVIEW

**Comment by minimalsm** [comment]
Hey @b17z, thanks for the PR here ❤️

Just a heads up: we won't have time to review these changes until probably next week. Thanks again 😉

**Comment by b17z** [comment]
Hey @minimalsm. Happy to try to provide some help!

And no problem!

File: src/content/developers/docs/consensus-mechanisms/index.md [hunk 1]
```diff
@@ -20,9 +20,36 @@ By consensus, we mean that a general agreement has been reached. Consider a grou
...
+* 02/03 Alice pays for Bob's boozy brunch. Total cost: $50. Bob owes Alice $25.
+* 02/07 Bob pays for Jason's groceries. Total cost: $200. Jason owes Bob $47.
+
+...
+
+Then, at the end of the month, the app will calculate total expenses and who owes who what money to settle up.
+
+Since all of you are rooomates, you should be able to trust each other that no one put a sneaky line item in to get one over an another rooomates.
+
+The problem with this system is when you start to expand the application to include more and more people. Alice invites 4 of her friends since she hangs out with those people more. Bob adds some of his family members because he's helping them with school expenses. With more people (who may or may not trust each other) and more throughput of line items added to the ledger (so, it's hard to keep track of it), how can anyone trust that the ledger is in the correct state after every line item? How can anyone trust that the ledger has not been tampered with? How can they all agree on how to settle up at the end of the month? Howe can they achieve consensus?
```
Comment (minimalsm) [inline src/content/developers/docs/consensus-mechanisms/index.md:46]: Whilst I agree with adding this, I don't think this is the right section for it. Any other thoughts/suggestions?

```suggestion
```

File: src/content/developers/docs/consensus-mechanisms/index.md [hunk 2]
```diff
@@ -84,6 +111,7 @@ The [Beacon Chain](/upgrades/beacon-chain/) uses a consensus mechanism called [C
 - [What is Nakamoto Consensus? Complete Beginner’s Guide](https://blockonomi.com/nakamoto-consensus/)
 - [How Does Casper work?](https://medium.com/unitychain/intro-to-casper-ffg-9ed944d98b2d)
 - [On the Security and Performance of Proof of Work Blockchains](https://eprint.iacr.org/2016/555.pdf)
+- [Byzantine fault & Byzanatine Generalss Problem](https://en.wikipedia.org/wiki/Byzantine_fault)
```
Comment (minimalsm) [inline src/content/developers/docs/consensus-mechanisms/index.md:114]:
```suggestion
- [Byzantine fault](https://en.wikipedia.org/wiki/Byzantine_fault)
```

File: src/content/developers/docs/consensus-mechanisms/index.md [hunk 3]
```diff
@@ -20,9 +20,36 @@ By consensus, we mean that a general agreement has been reached. Consider a grou
...
+
 ## What is a consensus mechanism? {#what-is-a-consensus-mechanism}
 
-Consensus mechanisms (also known as consensus protocols or consensus algorithms) allow distributed systems (networks of computers) to work together and stay secure.
+Consensus mechanisms (also known as consensus protocols or consensus algorithms<sup>1</sup>) allow distributed systems (networks of computers) to work together and stay secure.
+
+*<sup>1</sup>Note: Consensus protocols and consensus algorithms are often used interchangeably. However, protocols and algorithms are different. A protocol is a set of rules defined in a standard that govern how a system and its many functioning parts operate and interact. Algorithms are like exact recipes on how to solve a problem or calculate a result. Think of a car and how its many different parts dictating how the car should work (protocol) and the actions of a driver to get from point A to point B (algorithm)*
```
Comment (minimalsm) [inline src/content/developers/docs/consensus-mechanisms/index.md:52]: We can simplify here

```suggestion
Consensus protocols and consensus algorithms are often used interchangeably. However, protocols and algorithms are different. A protocol is a set of rules defined in a standard that governs how a system and its many functioning parts operate and interact. Algorithms are like exact recipes on how to solve a problem or calculate a result.
```

File: src/content/developers/docs/consensus-mechanisms/index.md [hunk 4]
```diff
@@ -20,9 +20,36 @@ By consensus, we mean that a general agreement has been reached. Consider a grou
...
+
+The problem with this system is when you start to expand the application to include more and more people. Alice invites 4 of her friends since she hangs out with those people more. Bob adds some of his family members because he's helping them with school expenses. With more people (who may or may not trust each other) and more throughput of line items added to the ledger (so, it's hard to keep track of it), how can anyone trust that the ledger is in the correct state after every line item? How can anyone trust that the ledger has not been tampered with? How can they all agree on how to settle up at the end of the month? Howe can they achieve consensus?
+
 ## What is a consensus mechanism? {#what-is-a-consensus-mechanism}
 
-Consensus mechanisms (also known as consensus protocols or consensus algorithms) allow distributed systems (networks of computers) to work together and stay secure.
+Consensus mechanisms (also known as consensus protocols or consensus algorithms<sup>1</sup>) allow distributed systems (networks of computers) to work together and stay secure.
```
Comment (minimalsm) [inline src/content/developers/docs/consensus-mechanisms/index.md:50]: Let's remove this, it doesn't follow a site convention for docs.

```suggestion
Consensus mechanisms (also known as consensus protocols or consensus algorithms) allow distributed systems (networks of computers) to work together and stay secure.
```

File: src/content/developers/docs/consensus-mechanisms/index.md [hunk 5]
```diff
@@ -20,10 +20,13 @@ By consensus, we mean that a general agreement has been reached. Consider a grou
 
 In regards to blockchain, the process is formalized, and reaching consensus means that at least 51% of the nodes on the network agree on the next global state of the network.
 
+
```
Comment (minimalsm) [inline src/content/developers/docs/consensus-mechanisms/index.md:23]:
```suggestion
```

**Review by minimalsm — APPROVED** [review]
Thanks for the PR ❤️ and sorry for the delay here! Merged my suggestions for the sake of getting this in but would like to continue the conversation on where the byzantine general's problem article should live.

**Comment by minimalsm** [comment]
@all-contributors please add @b17z for docs, ideas

**Pull request merged**